src/main/java/uk/gov/pay/directdebit/common/clients/GoCardlessClientFacade.java

```diff
@@ -50,12 +51,11 @@ public GoCardlessCustomer createCustomerBankAccount(MandateExternalId mandateExt
         return customer;
     }
 
-    public GoCardlessMandate createMandate(Mandate mandate, GoCardlessCustomer customer) {
-        com.gocardless.resources.Mandate gcMandate = goCardlessClientWrapper.createMandate(mandate.getExternalId(), customer);
-        return new GoCardlessMandate(
-                mandate.getId(),
-                GoCardlessMandateId.valueOf(gcMandate.getId()),
-                MandateBankStatementReference.valueOf(gcMandate.getReference()));
+    public PaymentProviderConfirmMandateResponse createMandate(Mandate mandate, GoCardlessCustomer customer) {
+        var createdMandate = goCardlessClientWrapper.createMandate(mandate.getExternalId(), customer);
```

Comment:
Avoiding having a huge fully-qualified class name in the type declaration by using `var`… sneaky! You have my admiration!